src/psyclone/tests/psyir/symbols/symbol_test.py

```diff
@@ -219,3 +223,81 @@ def test_argumentinterface_str():
 
     argument_interface = ArgumentInterface()
     assert str(argument_interface) == "Argument(pass-by-value=False)"
+
+
+def test_find_symbol_table():
+    ''' Test the find_symbol_table() method. '''
+    sym = Symbol("a_var")
+    with pytest.raises(TypeError) as err:
+        sym.find_symbol_table("3")
+    assert ("expected to be passed an instance of psyir.nodes.Node but got "
+            "'str'" in str(err.value))
+    # Search for a SymbolTable with only one level of hierarchy
+    sched = KernelSchedule("dummy")
+    table = sched.symbol_table
+    table.add(sym)
+    assert sym.find_symbol_table(sched) is table
+    # Create a Container so that we have two levels of hierarchy
+    ctable = SymbolTable()
+    sym2 = Symbol("b_var")
+    ctable.add(sym2)
+    _ = Container.create("test", ctable, [sched])
+    assert sym2.find_symbol_table(sched) is ctable
+    # A Symbol that isn't in any table
+    sym3 = Symbol("missing")
+    assert sym3.find_symbol_table(sched) is None
+    # When there is no SymbolTable associated with the PSyIR node
+    orphan = Literal("1", INTEGER_SINGLE_TYPE)
+    assert sym3.find_symbol_table(orphan) is None
+
+
+def test_symbol_copy():
+    '''
+    Test the Symbol.copy() method.
+    '''
+    csym = ContainerSymbol("some_mod")
+    asym = Symbol("a", visibility=Symbol.Visibility.PRIVATE,
+                  interface=GlobalInterface(csym))
+    new_sym = asym.copy()
+    assert new_sym is not asym
+    assert new_sym.name == asym.name
+    assert new_sym.interface == asym.interface
+    assert new_sym.visibility == asym.visibility
+
+
+def test_symbol_resolve_deferred():
```

Comment:
Since we have now the `get_external_symbol` public method, could you split the functionality of the new method in a separate explicit test for it. Then maybe you can simplify this one with a monkeypatch for `get_external_symbol`?

Also, can you remove the duplicated parts in `datasymbol_test.py::test_symbol_resolve_deferred` and also monkeypatch that test?

Reply:
Oh yes, that's much nicer now.